src/fontkit.cpp

```diff
@@ -146,27 +139,18 @@ extern "C" {
             if (FT_Load_Glyph(face, glyphId, FT_LOAD_RENDER)) {
                 ret = Val_error("[ERROR]: Unable to render character at FT_Load_Char\n");
             } else {
-                ReglfwImageInfo *p = (ReglfwImageInfo *)malloc(sizeof(ReglfwImageInfo));
-                p->numChannels = 1;
-                p->channelSize = 1;
-                p->width = face->glyph->bitmap.width;
-                p->height = face->glyph->bitmap.rows;
-
-                unsigned char *originalBuffer = face->glyph->bitmap.buffer;
-
-                int size = face->glyph->bitmap.width * face->glyph->bitmap.rows;
-                unsigned char* data = (unsigned char *)malloc(size);
-                memcpy(data, originalBuffer, size);
-
-                p->data = data;
+                FT_Bitmap bitmap = face->glyph->bitmap;
+                int bitmapDataLength = bitmap.rows * abs(bitmap.pitch);
+                bitmapBigarray = caml_ba_alloc_dims(CAML_BA_UINT8 | CAML_BA_C_LAYOUT, 2, NULL, bitmap.rows, abs(bitmap.pitch));
+                memcpy(Caml_ba_data_val(bitmapBigarray), (const char *)bitmap.buffer, bitmapDataLength);
```

Comment:
Excellent! That's great that we can copy directly into the Bigarray buffer vs allocated an extra one. Nice find 👍